### PR TITLE
Conveyor belt QoL

### LINKED ISSUE #8727: Make conveyor belts copy your dir when built somehow
Please

### PR DESCRIPTION
**Pull request by 9600bauds**
Reworked conveyor belt construction to be faster and simpler: Conveyor belts are now built with their own machine frame which doesn't require wiring and can be walked over. This new conveyor belt frame will also copy the direction you're facing when building it. To make a conveyor belt assembly: Build it from 5 metal sheets, just like a machine frame.
Conveyor belt assembly sprites courtesy of Blithering.
Newly built conveyor belts will now automatically try to copy the frequency and ID tag of any other conveyor belts immediately next to them. This lets you easily create long conveyor belt lines without having to configure every single one of them, so long as you configured the first one before building the others.
Multitool buffer cloning now works on conveyor belts: Add a conveyor belt/conveyor switch to a Multitool's buffer, click the Multitool in your hand to enable cloning mode, then click on a conveyor belt. It will instantly copy the frequency and ID tag from buffer, and if the buffer was taken from a conveyor belt, it will also copy it's direction.

Also added a helpful little snippet of instructions to the conveyor belt multitool menu.

Fixes #8727 

Webm: http://a.pomf.cat/eybskk.webm

### CODE REVIEW

**Comment by ghost** [comment]
It doesn't really sound that intuitive from what you're saying.

**Comment by 9600bauds** [comment]
Which part
Webm soon

**Comment by 9600bauds** [comment]
http://a.pomf.cat/idzhsq.webm
Seems pretty intuitive to me, it's definitely a HUGE improvement over having to set every single conveyor belt section's direction and ID tag from the menu

**Comment by unid15** [comment]
What do you use to record webms, @9600bauds?

File: code/game/machinery/constructable_frame.dm [hunk 1]
```diff
@@ -156,6 +156,8 @@
 							else
 								circuit.loc = null
 							new_machine.RefreshParts()
+							if(istype(new_machine, /obj/machinery/conveyor)) //unfortunately I can't do this on the conveyor belt's New() or something actually OOP because I need the user
```
Comment (PJB3005) [inline code/game/machinery/constructable_frame.dm]: There are so much ways to approach this, for example at the circuit board level.

Don't do this.

**Comment by 9600bauds** [comment]
For recording I use Shadowplay which is great because of the "record constantly but only keep the last 2 minutes in cache so if anything funny happens i press a button and have the last 2 minutes saved thus I will never miss anything" feature, to convert them I use Webm For Retards and replace -crf 32 in the advanced arguments with -crf 16 -qmin 0 -qmax 35

File: code/game/machinery/constructable_frame.dm (hunk 1 above)
Comment (9600bauds) [inline code/game/machinery/constructable_frame.dm]: But how? I need to read the user's dir at the LAST step of the construction

Comment (PJB3005) [inline code/game/machinery/constructable_frame.dm]:
``` DM
/obj/item/weapon/circuitboard/finish_build(var/obj/machinery/new_machine, var/mob/user)
```

You can fill in the rest I assume.

Comment (9600bauds) [inline code/game/machinery/constructable_frame.dm]: Oh yes, that's good

**Comment by 9600bauds** [comment]
There now it's good

**Comment by Iamgoofball** [comment]
conveyor belts are useless to build until you put them on its own ticker at double or triple speed

**Comment by 9600bauds** [comment]
I like them, shut up goofball

**Comment by unid15** [comment]
but we already have that goodball

**Comment by Exxion** [comment]
Copying your direction doesn't really seem that intuitive
However, making multitool cloning work on them is a big good

**Comment by 9600bauds** [comment]
How is it not intuitive? I don't get it
You can see me sokoban the whole conveyor belt's flow by just running into the frames
Before they would always point up

**Comment by Iamgoofball** [comment]
oh did you guys port that?

if so gj, it's good shit

**Comment by PJB3005** [comment]
@Iamgoofball I did.

**Comment by 9600bauds** [comment]
I thought of something better: Why not just copy the ID tag from any conveyor belts within 1 tile on New()?

**Comment by Imagebard** [comment]
Multitooling them with 'clone' is easy enough, why would you need that?

**Comment by 9600bauds** [comment]
Look at the webm: I need to multitool clone ALL OF THEM after constructing them, if they cloned themselves on New() I would only need to set up one of them and the rest would configure themselves like dominos

**Comment by Imagebard** [comment]
I can only preview this causing more trouble for someone that wants to do something more complex with Conveyor Belts.
I still recommend you don't do it.

**Comment by Exxion** [comment]
It would make it much easier to make simple conveyor setups and would have exactly no effect on the difficulty of complex ones

**Comment by 9600bauds** [comment]
How so? The other option is that they are not configured at all

**Comment by clusterfack** [comment]
Oh I didn't see this, seems you already did something similar to my suggestion. Meh I don't like that it's built with a machine frame anyway, takes 5 extra seconds needlessly.

**Comment by 9600bauds** [comment]
Yep, agreed. Would it be possible to eschew the machine frame and do the assembly thing you said while still keeping it a obj/machinery?

**Comment by clusterfack** [comment]
Yes it would

**Comment by 9600bauds** [comment]
I'd have to try that sometime, but I would need someone to sprite the assembly so rip

**Comment by 9600bauds** [comment]
Behold: Even more quality!

**Comment by 9600bauds** [comment]
I plan to work on this a little more

**Comment by clusterfack** [comment]
Factorio style conveyors when

**Comment by 9600bauds** [comment]
I haven't played that game, what's special about them

**Comment by clusterfack** [comment]
Nah it just perfectly illustrated to me how I wanted conveyor construction to work here, a stack of conveyor assemblies you can place on the ground really easily and change the direction of while they're in your hand.

**Comment by 9600bauds** [comment]
The way I'm making them is that you craft the assemblies from metal, and they copy your direction when they appear under you. I dunno how they would be stackable though

**Comment by 9600bauds** [comment]
By making them from metal I mean the same way machine frames work, just non-dense

**Comment by clusterfack** [comment]
Like you craft a whole stack of assemblies from components, but I'm just going off on a tangent here because I found factorio's system pretty enjoyable in the short time I played it

**Comment by ghost** [comment]
We should have factorio style mining

**Comment by clusterfack** [comment]
The current system is so blah I'd be down. Tiers don't even exist right now

**Comment by 9600bauds** [comment]
Behold the new conveyor belt assembly sprites courtesy of @Blithering
http://a.pomf.cat/eybskk.webm

**Comment by 9600bauds** [comment]
This touches up stack construction to be more OOP so that's worth looking over

**Comment by 9600bauds** [comment]
Aaaaahhhh stop modifying sheet_types.dm

**Comment by 9600bauds** [comment]
Since nobody said anything about the OOP I added to stack construction I'm going to assume it's OK

**Comment by 9600bauds** [comment]
Insidious NFD because jesus fuck it's been a while and I don't want to fix any more conflicts on this thing

**Comment by Imagebard** [comment]
Okay, form what I know, NFD means a high-positioning moderator needs to approve or disapprove of the changes.
Why would this PR need that? Does it affect game balance or something I didn't notice?

**Comment by gbasood** [comment]
NFD, recently, has been used more along the lines of "I'm a tagman and I fixed all my shit that I know of, someone make sure it's good and merge it already"

**Comment by Exxion** [comment]
It's a tag to say "Hey someone merge this please"
It _used_ to be that pretty much everything had to have it and then Pomf had to merge things, but Pomf's not around enough anymore for that to work.

**Comment by ihadtoregisterforthis** [comment]
Still holier than self-merging, at least you share the blame if something is messed up

**Pull request merged**